common/djangoapps/student/tests/test_models.py

```diff
@@ -713,6 +714,114 @@ def test_not_enrolled_student_is_enrolled(self):
         assert actual_student.is_active is True
         assert actual_cea.user == student
 
+    @mock.patch.dict('django.conf.settings.FEATURES', {'DISABLE_AUTOENROLL_AFTER_COURSE_END': True})
+    def test_student_not_enrolled_if_course_ended_and_flag_true(self):
+        """
+        Make sure that invited students who are not enrolled does not get
+        auto enrolled when their account is activated if the course has ended
+        and the DISABLE_AUTOENROLL_AFTER_COURSE_END is set to True
+        """
+
+        # Set end date to a past date so that course is ended
+        start_date = datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=3)
+        end_date = datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=2)
+        course_overview = CourseOverviewFactory.create(
+            id=self.course.id,
+            start=start_date,
+            end=end_date,
+        )
+        course_overview.save()
+
+        # fixture
+        student = self._set_up_invited_student(
+            course=self.course,
+            active=False,
+            enrolled=False
+        )
+
+        # # trigger the post_save callback
+        student.is_active = True
+        student.save()
+
+        actual_cea = CourseEnrollmentAllowed.objects.get(email=student.email)
+
+        assert actual_cea.user is None
+
+    @mock.patch.dict('django.conf.settings.FEATURES', {'DISABLE_AUTOENROLL_AFTER_COURSE_END': False})
```

Comment:
This is the default behavior, so we shouldn't need this override. This way, we will be also testing that the default behavior is unchanged.

Reply:
Makes sense. Now testing with default value.